src/expr/node_algorithm.cpp

```diff
@@ -118,6 +127,60 @@ bool hasSubtermMulti(TNode n, TNode t)
   return false;
 }
 
+bool hasSubterm(TNode n, const std::vector<Node>& t, bool strict)
+{
+  if (t.empty())
+  {
+    return false;
+  }
+  if (!strict && std::find(t.begin(), t.end(), n) != t.end())
+  {
+    return true;
+  }
+
+  std::unordered_set<TNode, TNodeHashFunction> visited;
+  std::vector<TNode> toProcess;
+
+  toProcess.push_back(n);
+
+  for (unsigned i = 0; i < toProcess.size(); ++i)
```

Comment:
for each?

Reply:
`toProcess` changes over the course of the execution of the function actually. Does the for each iteration work in this case?

Reply:
Oh I see, in this case leave it, but maybe add a comment so that it's clear when you look at the for loop header.